### PR TITLE
Fix articles showing cookie information in reader mode

### PR DESCRIPTION
**Pull request by tthhtao**
Applying the `aria-modal` property to an element with `role="dialog"` replaces the technique of using `aria-hidden` in the background. So I think this kind of element should also be considered "invisible" to the reader.

### CODE REVIEW

File: Readability.js [hunk 1]
```diff
@@ -904,6 +904,12 @@ Readability.prototype = {
           continue;
         }
 
+        // User is not able to see elements applied with both "aria-modla = true" and "role = dialog"
+        if (node.getAttribute("aria-modal") == "true" && node.getAttribute("role") == "dialog") {
+          node = this._removeAndGetNext(node);
+          continue;
+        }
```
Comment (tigleym) [inline Readability.js:911]: This should be added to the check in [isProbablyVisible](https://github.com/tthhtao/readability/blob/2edee0ee6224b9c0ed94f81832cb5e2678d0f0d4/Readability.js#L2206).

**Review by tigleym — CHANGES_REQUESTED** [review]
Thanks @tthhtao ! This works nicely, but I think we should address the comment I have here.

File: Readability.js [hunk 2]
```diff
@@ -2202,7 +2208,9 @@ Readability.prototype = {
     return (!node.style || node.style.display != "none")
       && !node.hasAttribute("hidden")
       //check for "fallback-image" so that wikimedia math images are displayed
-      && (!node.hasAttribute("aria-hidden") || node.getAttribute("aria-hidden") != "true" || (node.className && node.className.indexOf && node.className.indexOf("fallback-image") !== -1));
+      && (!node.hasAttribute("aria-hidden") || node.getAttribute("aria-hidden") != "true" || (node.className && node.className.indexOf && node.className.indexOf("fallback-image") !== -1))
+      // user can not see element applied with both attribute "aria-modla = true" and "role = dialog"
```
Comment (tigleym) [inline Readability.js:2212]: "arial-modal"

File: Readability.js (hunk 1 above)
Comment (tigleym) [reply Readability.js:911]: We should remove this check since it should be handled in the one above now

**Review by tigleym — CHANGES_REQUESTED** [review]

**Comment by gijsk** [comment]
I think all the review comments were addressed so let's get this merged.

**Comment by gijsk** [comment]
@tthhtao sorry this sat for such a long time!

**Pull request merged**